fastembed/sparse/splade_pp.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Dict, Iterable, List, Optional, Tuple, Union
+
+import numpy as np
+
+from fastembed.common.onnx_model import OnnxModel, EmbeddingWorker
+from fastembed.common.utils import define_cache_dir
+from fastembed.sparse.sparse_embedding_base import SparseEmbedding, SparseTextEmbeddingBase
+
+supported_splade_models = [
+    {
+        "model": "prithvida/SPLADE_PP_en_v1",
+        "vocab_size": 30522,
+        "description": "Independent Implementation of SPLADE++ Model for English",
+        "size_in_GB": 0.532,
+        "sources": {
+            "hf": "nirantk/SPLADE_PP_en_v1",
+        },
+    },
+]
+
+
+class SpladePP(SparseTextEmbeddingBase, OnnxModel[SparseEmbedding]):
+
+    @classmethod
+    def _post_process_onnx_output(cls, output: Tuple[np.ndarray, np.ndarray]) -> Iterable[SparseEmbedding]:
+        logits, attention_mask = output
+        relu_log = np.log(1 + np.maximum(logits, 0))
+
+        weighted_log = relu_log * np.expand_dims(attention_mask, axis=-1)
+
+        max_val = np.max(weighted_log, axis=1)
+
+        # Score matrix of shape (batch_size, vocab_size)
+        # Most of the values are 0, only a few are non-zero
+        scores = np.squeeze(max_val)
+        for row_scores in scores:
+            indices = row_scores.nonzero()[0]
+            scores = row_scores[indices]
```

Comment:
I'd not want to change the user expectation to getting indices and scores. A binary vector has all the information a user needs to get the indices and scores. What is the advantage of doing this for the user?

Reply:
You mean you would prefer "densified" version of sparse vector? It won't work with all other potential models we could introduce. E.g. BM25 have much bigger vocab, we can't simply put into numpy array.

Additionally current format is how qdrant works with sparse vectors, so it should be easy to interact with.
We can add built-in conversion back into numpy, if you need this representation for some reason.